sdl_android/src/main/java/com/smartdevicelink/protocol/WiProProtocol.java

```diff
@@ -172,57 +172,59 @@ else if (protocolMsg.getBulkData() != null) {
 		}
 		
 		synchronized(messageLock) {
-			if (data.length > MAX_DATA_SIZE) {
-				
-				messageID++;
-	
-				// Assemble first frame.
-				int frameCount = data.length / MAX_DATA_SIZE;
-				if (data.length % MAX_DATA_SIZE > 0) {
-					frameCount++;
-				}
-				//byte[] firstFrameData = new byte[HEADER_SIZE];
-				byte[] firstFrameData = new byte[8];
-				// First four bytes are data size.
-				System.arraycopy(BitConverter.intToByteArray(data.length), 0, firstFrameData, 0, 4);
-				// Second four bytes are frame count.
-				System.arraycopy(BitConverter.intToByteArray(frameCount), 0, firstFrameData, 4, 4);
-
-				SdlPacket firstHeader = SdlPacketFactory.createMultiSendDataFirst(sessionType, sessionID, messageID, _version,firstFrameData,protocolMsg.getPayloadProtected());
-				firstHeader.setPriorityCoefficient(1+protocolMsg.priorityCoefficient);
-				//Send the first frame
-				handlePacketToSend(firstHeader);
-				
-				int currentOffset = 0;
-				byte frameSequenceNumber = 0;
-
-				for (int i = 0; i < frameCount; i++) {
-					if (i < (frameCount - 1)) {
-	                     ++frameSequenceNumber;
-	                        if (frameSequenceNumber ==
-	                                SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME) {
-	                            // we can't use 0x00 as frameSequenceNumber, because
-	                            // it's reserved for the last frame
-	                            ++frameSequenceNumber;
-	                        }
-					} else {
-						frameSequenceNumber = SdlPacket.FRAME_INFO_FINAL_CONNESCUTIVE_FRAME;
-					} // end-if
-					
-					int bytesToWrite = data.length - currentOffset;
-					if (bytesToWrite > MAX_DATA_SIZE) { 
-						bytesToWrite = MAX_DATA_SIZE; 
+			if (data != null) {

```

Comment:
I see. So there are 2 ways we can handle this now.
1. When data is null, don't send anything (fix stays the same) and add documentation to the callers of the method.
2. When data is null, do send the packet with null data like so (similar to the case where data.length < MAX_DATA_SIZE):
`SdlPacket header = SdlPacketFactory.createSingleSendData(sessionType, sessionID, data.length, messageID, _version, data, protocolMsg.getPayloadProtected());`
and replace:
a. `data.length` with either MAX_DATA_SIZE or negative.
b. data = data (null)
Internally when the code tries to create a packet in `SdlPacket.java`, `constructPacket()` method, the null data is not added to the ByteBuffer due to this check:
`if(payload!=null && payload.length>0){
			builder.put(payload);
		}
`
If there is no imperative need to send the packet when data is null, I would vote for 1.